### ISSUE TITLE
Windows Terminal Crashed on window resizing.

### ISSUE THREAD

**Issue by kxxt** [microsoft/terminal#8596]
# Environment

```none
Windows build number: 10.0.19042.0
Windows Terminal version (if applicable): 1.4.3243.0

Any other software?
I was running mongod and python shell inside Windows Terminal.(However I think the problem is not related to them).
```

# Steps to reproduce

I docked Windows Terminal to the right half of my screen (another app to the left half) , and then I resized the window width (dragging the left edge of the window) which caused the app crash.
However, I failed to duplicate the case.

# Expected behavior

The window should  be resized without crashing

# Actual behavior

The app crashed.

# Additional Information

[Link to Feedback hub](https://aka.ms/AAan0tz)

There are two screens : 
A screen: 1080p,main
B screen: 4K.

Before crashing , Windows Terminal is on the B screen.

## From Diagnostic Data Viewer
```
Description
Faulting Application Path:	 C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.4.3243.0_x64__8wekyb3d8bbwe\WindowsTerminal.exe
Creation Time:			 2020/12/16 8:54:38
Problem:			 已停止工作
Status:				 报告已发送

Problem signature
Problem Event Name:           MoAppCrash
程序包全名:                        Microsoft.WindowsTerminal_1.4.3243.0_x64__8wekyb3d8bbwe
应用程序名:                        praid:App
应用程序版本:                       1.4.2011.19003
应用程序时间戳:                      5fb70b02
故障模块名称:                       combase.dll
故障模块版本:                       10.0.19041.662
故障模块时间戳:                      974bdb64
异常代码:                         800700e8
异常偏移:                         000000000006ea56

Extra information about the problem
Bucket ID:			 7df1f4f4b279fc0f31b964d5bb1d57f9 (1277162838466320377)

```

Crash Dump:

[WindowsTerminal.exe.19888.dmp.zip](https://github.com/microsoft/terminal/files/5700174/WindowsTerminal.exe.19888.dmp.zip)

**Comment by kxxt** [comment]
## Other data from Diagnostic data viewer.
```
{
    "ver": "4.0",
    "name": "Microsoft.Windows.FaultReporting.AppCrashEvent",
    "time": "2020-12-16T00:54:33.5805208Z",
    "iKey": "o:0a89d516ae714e01ae89c96d185e9ae3",
    "ext": {
        "utc": {
            "shellId": 33786497709047811,
            "eventFlags": 258,
            "pgName": "WINCORE",
            "flags": 905970224,
            "epoch": "48200357",
            "seq": 138
        },
        "metadata": {
            "f": {
                "ProcessCreateTime": 9,
                "ExceptionOffset": 5,
                "AppVersion": 5,
                "ModVersion": 5,
                "AppSessionGuid": 8,
                "TargetAsId": 5
            },
            "privTags": 16777216
        },
        "os": {
            "bootId": 482,
            "name": "Windows",
            "ver": "10.0.19042.685.amd64fre.vb_release.191206-1406",
            "expId": "FX:118B0639,FX:11A8C293,FX:11A8C2FE,FX:11C0E96C,FX:11C3F9D5,FX:11C7EB6A,FX:11C940DB,FX:11C96C7A,FX:11CB9A9F,FX:11CB9AC1,FX:11D898D7,FX:1A92E6F,FX:1D5BFCD"
        },
        "app": {
            "id": "W:0000f519feec486de87ed73cb92d3cac802400000000!00007c6560566535e966989f80e45fc3fef426b0b48a!werfault.exe",
            "ver": "2056/12/18:01:15:28!92356!werfault.exe",
            "asId": 2110
        },
        "device": {
            "localId": "s:0D0C1270-0049-4809-BBBC-4086A783871E",
            "deviceClass": "Windows.Desktop"
        },
        "protocol": {
            "devMake": "HP",
            "devModel": "HP Spectre x360 Convertible 15-eb0xxx"
        },
        "user": {
            "localId": "w:F3821754-1012-4775-0ADB-A748ACA4E4E2"
        },
        "xbl": {
            "sbx": "RETAIL",
            "did": "F900EF63B801727E",
            "xid": "x:2535430858658376"
        },
        "loc": {
            "tz": "+08:00"
        }
    },
    "data": {
        "ProcessId": 14584,
        "ProcessCreateTime": "2020-12-16T00:34:35.7422227Z",
        "ExceptionCode": 3221226505,
        "ExceptionOffset": 1097052,
        "AppName": "OpenConsole.exe",
        "AppVersion": 5348877619612090369,
        "AppTimeStamp": 1605831363,
        "ModName": "KERNELBASE.dll",
        "ModVersion": 186418215054737418,
        "ModTimeStamp": 3965251605,
        "PackageFullName": "Microsoft.WindowsTerminal_1.4.3243.0_x64__8wekyb3d8bbwe",
        "PackageRelativeAppId": "App",
        "ProcessArchitecture": 9,
        "ReportId": "9d20a265-7c6c-4750-8d67-f8a0647d003e",
        "Flags": 2147483648,
        "AppSessionGuid": "000038F8-0001-01E2-93A0-D63A43D3D601",
        "TargetAppId": "U:Microsoft.WindowsTerminal_1.4.3243.0_x64__8wekyb3d8bbwe!App",
        "TargetAppVer": "1.4.3243.0_x64_!2020/11/20:00:16:03!10af09!OpenConsole.exe",
        "TargetAsId": 1673,
        "FriendlyAppName": "OpenConsole.exe",
        "IsFatal": true
    }
}
```
```
{
    "ver": "4.0",
    "name": "Microsoft.Windows.FaultReporting.AppCrashEvent",
    "time": "2020-12-16T00:54:38.2977854Z",
    "iKey": "o:0a89d516ae714e01ae89c96d185e9ae3",
    "ext": {
        "utc": {
            "shellId": 33786497709047811,
            "eventFlags": 258,
            "pgName": "WINCORE",
            "flags": 905970224,
            "epoch": "48200357",
            "seq": 139
        },
        "metadata": {
            "f": {
                "ProcessCreateTime": 9,
                "ExceptionOffset": 5,
                "AppVersion": 5,
                "ModVersion": 5,
                "AppSessionGuid": 8,
                "TargetAsId": 5
            },
            "privTags": 16777216
        },
        "os": {
            "bootId": 482,
            "name": "Windows",
            "ver": "10.0.19042.685.amd64fre.vb_release.191206-1406",
            "expId": "FX:118B0639,FX:11A8C293,FX:11A8C2FE,FX:11C0E96C,FX:11C3F9D5,FX:11C7EB6A,FX:11C940DB,FX:11C96C7A,FX:11CB9A9F,FX:11CB9AC1,FX:11D898D7,FX:1A92E6F,FX:1D5BFCD"
        },
        "app": {
            "id": "W:0000f519feec486de87ed73cb92d3cac802400000000!00007c6560566535e966989f80e45fc3fef426b0b48a!werfault.exe",
            "ver": "2056/12/18:01:15:28!92356!werfault.exe",
            "asId": 2116
        },
        "device": {
            "localId": "s:0D0C1270-0049-4809-BBBC-4086A783871E",
            "deviceClass": "Windows.Desktop"
        },
        "protocol": {
            "devMake": "HP",
            "devModel": "HP Spectre x360 Convertible 15-eb0xxx"
        },
        "user": {
            "localId": "w:F3821754-1012-4775-0ADB-A748ACA4E4E2"
        },
        "xbl": {
            "sbx": "RETAIL",
            "did": "F900EF63B801727E",
            "xid": "x:2535430858658376"
        },
        "loc": {
            "tz": "+08:00"
        }
    },
    "data": {
        "ProcessId": 19888,
        "ProcessCreateTime": "2020-12-16T00:34:35.2243240Z",
        "ExceptionCode": 3221226107,
        "ExceptionOffset": 1097052,
        "AppName": "WindowsTerminal.exe",
        "AppVersion": 5348877619612090369,
        "AppTimeStamp": 1605831426,
        "ModName": "KERNELBASE.dll",
        "ModVersion": 186418215054737418,
        "ModTimeStamp": 3965251605,
        "PackageFullName": "Microsoft.WindowsTerminal_1.4.3243.0_x64__8wekyb3d8bbwe",
        "PackageRelativeAppId": "App",
        "ProcessArchitecture": 9,
        "ReportId": "02261e47-8048-45a6-b4ae-6f4b84f669f7",
        "Flags": 2147483648,
        "AppSessionGuid": "00004DB0-0001-01E2-289A-873A43D3D601",
        "TargetAppId": "U:Microsoft.WindowsTerminal_1.4.3243.0_x64__8wekyb3d8bbwe!App",
        "TargetAppVer": "1.4.3243.0_x64_!2020/11/20:00:17:06!37735!WindowsTerminal.exe",
        "TargetAsId": 1667,
        "FriendlyAppName": "WindowsTerminal.exe",
        "IsFatal": true
    }
}
```

**Comment by zadjii-msft** [comment]
Huh. Bizarre that `OpenConsole.exe` was the thing to crash. That at least narrows the source of the issue down considerably. Was text being actively output to the Terminal when you tried resizing? Did you possible have anything typed in at a prompt line (but hadn't yet hit <kbd>enter</kbd>)?

**Comment by Don-Vito** [comment]
@zadjii-msft - are the terminal symbol files published somewhere? This could make dump analysis much more feasible :smile:

**Comment by kxxt** [comment]
> Huh. Bizarre that `OpenConsole.exe` was the thing to crash. That at least narrows the source of the issue down considerably. Was text being actively output to the Terminal when you tried resizing? Did you possible have anything typed in at a prompt line (but hadn't yet hit <kbd>enter</kbd>)?

@zadjii-msft a MongoDB server was running inside first tab, and it might be outputing something.
Maybe something was typed in the second tab(running python,I didn't remember).

**Comment by zadjii-msft** [comment]
(as a mental note, there was also #8287 recently, though I'm not sure we ever really dug into the cabs to see what the root cause was)

**Comment by Lightjohn** [comment]
Hi, not sure I can be of help but I always get instant (segfault?) crash of terminal mostly when I have splitted in multiple terminal and one/both of them I am ssh ing on my Linux server it happens often enough that I forbid myself to resize if there is anything important.

Sadly I don't know how I could give you more information on what is phappenign precisely.

One sure was for me to reproduce this error is:
* open WSL Ubuntu 20.04 terminal
* split horizontally to a second terminal
* In the first one ssh linux server then open htop
* Drag to resize the windows and it wil crash 100% of time

**Comment by zadjii-msft** [comment]
Uhg. Well, unfortunately, there's nothing useful in this dump, or feedback hub link. No cabs, and the dump just points at some unknown XAML stack north of `Windows_UI_Xaml!GetErrorContextIndex`, with no info in the xaml stowed exception 😢 

I'm gonna guess based on the info available that this is the same crash as #9547, which is #7417, which is #4907. 

/dup #4907
